src/mapGenerator/mapgenManager.py

```diff
@@ -0,0 +1,156 @@
+# system imports
+import logging
+import sys
+from PyQt5 import QtCore, QtGui, QtWidgets, QtNetwork
+import os
+import re
+import shutil
+import random
+
+# local imports
+from config import Settings
+import fafpath
+from vault.dialogs import downloadFile
+from mapGenerator.mapgenProcess import MapGeneratorProcess
+from fa.maps import getUserMapsFolder
+
+logger = logging.getLogger(__name__)
+
+class MapGeneratorManager(object):
+    def __init__(self):
+        self.releaseUrl = "https://github.com/FAForever/Neroxis-Map-Generator/releases/download/{}/NeroxisGen_{}.jar"
+        self.latestRelease = "https://github.com/FAForever/Neroxis-Map-Generator/releases/latest"
+        self.latestVersion = None
+        self.response = None
+
+        self.currentVersion = Settings.get('mapGenerator/version', "0", str)
+        self.previousMaps = Settings.get('mapGenerator/deleteMaps')
+        self.generatorPath = os.path.join(fafpath.get_libdir(), "map-generator")
+        self.mapsFolder = getUserMapsFolder()
+
+        if self.previousMaps:
+            self.deletePreviousMaps()
+
+    def generateMap(self, mapname):
+        dataString =  mapname.split("neroxis_map_generator",1)[1]
+        version = re.search(r'\_(.*?)\_',dataString).group(1)
```

Comment:
please use a full map name regex pattern (see java client) and match against the entire map name.